cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/monitor/RuntimeMonitor.java

```diff
@@ -191,14 +199,29 @@ protected long runTask() throws Exception {
       monitorResponses = monitorClient.fetchMessages(topicsToRequest);
     } catch (ServiceUnavailableException | IOException e) {
       OUTAGE_LOGGER.warn("Failed to fetch monitoring messages for program {}", programRunId, e);
+      numConsecutivePollFailures++;
```

Comment:
This catch block actually doesn't catch all exception. Suggest to consolidate the logic it in the `shouldRetry` method instead.